browser/brave_ads/ads_service_impl.cc

```diff
@@ -2098,6 +2106,96 @@ void AdsServiceImpl::RecordP2AEvent(const std::string& name,
   }
 }
 
+void AdsServiceImpl::LogTrainingCovariates(
```

Comment:
Training covariates could be decoupled from `AdsImpl`, i.e. `training_covariates.h/.cc` with a LogTrainingCovariate method. We can then raise a follow up issue to look at decoupling the switch into a factory

Reply:
as discussed not working it for now - if ads imp should be broken up